include/fcl/narrowphase/detail/convexity_based_algorithm/gjk_libccd-inl.h

```diff
@@ -1005,6 +1005,130 @@ static int __ccdEPA(const void *obj1, const void *obj2,
     return 0;
 }
 
+/** Returns the points p1 and p2 on the original shapes that correspond to point
+ * p in the given simplex.*/
+static void extractClosestPoints(ccd_simplex_t* simplex,
+                                 ccd_vec3_t* p1, ccd_vec3_t* p2, ccd_vec3_t* p)
+{
+  int simplex_size = ccdSimplexSize(simplex);
+  assert(simplex_size <= 3);
+  if (simplex_size == 1)
+  {
+      // Closest points are the ones stored in the simplex
+      if(p1) *p1 = simplex->ps[simplex->last].v1;
+      if(p2) *p2 = simplex->ps[simplex->last].v2;
+  }
+  else if (simplex_size ==2)
+  {
+      // Closest points lie on the segment defined by the points in the simplex
+      // Let the segment be defined by points A and B. We can write p as
+      //
+      // p = A + s*AB, 0 <= s <= 1
+      // p - A = s*AB
+      //
+      // This defines three equations, but we only need one. Take the x
+      // component
```

Comment:
The quality of this solution degrades as the line segment becomes more perpendicular to the x-axis (i.e., where `B_x - A_x << 1`. Ideally, you'd want to pick the axis where `|B_i - A_i|` is biggest.